tests/unit/mainwindow/test_prompt.py

```diff
@@ -94,6 +96,33 @@ def test_backspacing_path(self, qtbot, tmp_path, get_prompt):
         prompt.item_focus('next')
         assert prompt._lineedit.text() == str(testdir / 'bar')
 
+    @pytest.mark.parametrize("keys,expected",
+                            [([], ['bar', 'foo', 'bat'].sort()),
+                            (['F'], ['foo'].sort()),
+                            (['Backspace', 'A'], ['bar', 'bat'].sort())])
+    def test_filtering_path(self, qtbot, tmp_path, get_prompt, keys, expected):
+        testdir = tmp_path / 'test'
+
+        for directory in ['bar', 'foo', 'bat']:
+            (testdir / directory).mkdir(parents=True)
+
+        prompt = get_prompt(str(testdir) + os.sep)
+
+        for _ in range(len(keys)):
+            for key in keys:
+                eval("qtbot.keyPress(prompt._lineedit, Qt.Key_{})".format(key))
```

Comment:
Eval is a common misspelling of evil :wink:

Just use the `Qt.Key_...` constants in the parametrization instead, makes this much simpler, and it's not like the `Qt.Key_` prefix is overly long.

If you had to do this for some reason, you could use `value = getattr(Qt, f'Key_{key}')` to avoid the eval, but I'd prefer keeping things simple and just using the enum numbers in the decorator.